File: algorithms/Strings & Searching/binary_search/binary_search.py
#BINARY SEARCH IN AN ARRAY...............

#__________________________________________________________________________________
class abc(object):
    def __init__(self):
        self.n=int(input("Enter no of items in the list: "))
        self.l=[]
        self.take()

    def take(self):
        for i in range(self.n):
            self.l+=[int(input("Enter the value: "))]

        print ("The list u entered is: ",self.l)
        print()
        self.l.sort()
        print ("The sorted list is: ",self.l)
        self.search()

    def search(self):
        flag=0
        a=int(input("Enter the value to search: "))
        beg=0
        end=self.n-1
        while beg<end:
            mid=(beg+end)//2
            if self.l[mid]>a:
                end=mid-1
            elif self.l[mid]<a:
                beg=mid+1
            elif self.l[mid]==a:
                print ("Search item found.")
                print ("Found at ",mid+1," position.")
                flag=1
                break
        if flag==0:
            print ("Search unsuccesfull.")

obj=abc()

File: algorithms/Searching/binary_search/binary_search.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
	int N, index = 0, index1 = 0, index2 = 0, key, flag = 0, beg, end, mid;

	cout << "Enter the no of items in the list: ";
	cin >> N;

	int arr[N];
	beg = 0;
	end = N - 1;

	while(index < N)
	{
		cout << "Enter the value: ";
		cin >> arr[index];
		index++;
	}

	cout << "Enter the value to search: ";
	cin >> key;

	cout << "The list you entered is: [";
	while(index1 < N-1)
	{
		cout << arr[index1] << ", ";
		index1++;
	}
	cout << arr[index1] << "]" << endl;

	sort(arr, arr + N);

	cout << "The sorted list is: [";
	while(index2 < N-1)
	{
		cout << arr[index2] << ", ";
		index2++;
	}
	cout << arr[index2] << "]" << endl;

	while(beg < end)
	{
		mid = (beg + end) / 2;

		if(arr[mid] == key)
		{
			end = 0;
			flag = 1;
		}
		else if(arr[mid] > key)
		{
			end = mid - 1;
		}
		else if(arr[mid] < key)
		{
			beg = mid + 1;
		}
	}

	if(flag == 1)
	{
		cout << "Search Successful." << endl;
	}
	else
	{
		cout << "Search Unsuccessful." << endl;
	}

	return 0;
}